test/model/autocomplete_test.dart

```diff
@@ -349,4 +320,131 @@ void main() {
       doCheck('Four F', eg.user(fullName: 'Full Name Four Words'), false);
     });
   });
+
+  group('MentionAutocompleteView sorting users results', () {
+    late PerAccountStore store;
+
+    Future<void> prepare({
+      List<User> users = const [],
+      List<RecentDmConversation> dmConversations = const [],
+    }) async {
+      store = eg.store(initialSnapshot: eg.initialSnapshot(
+        recentPrivateConversations: dmConversations));
+      await store.addUsers(users);
+    }
+
+    group('MentionAutocompleteView.compareByDms', () {
+      test('has DMs with userA and userB, latest with userA, prioritizes userA', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [1],    maxMessageId: 200),
+          RecentDmConversation(userIds: [1, 2], maxMessageId: 100),
+        ]);
+
+        final userA = eg.user(userId: 1);
+        final userB = eg.user(userId: 2);
+        final compareAB = MentionAutocompleteView.compareByDms(userA, userB, store: store);
+        check(compareAB).isNegative();
+      });
+
+      test('has DMs with userA and userB, latest with userB, prioritizes userB', () async {
+        await prepare(dmConversations: [
+          RecentDmConversation(userIds: [2],    maxMessageId: 200),
+          RecentDmConversation(userIds: [1, 2], maxMessageId: 100),
```

Comment:
There's a bit of a mismatch in the naming here. The descriptions call the users "A" and "B":
```dart
      test('has DMs with userA and userB, latest with userA, prioritizes userA', () async {
```
as do the `User` locals (though my suggestion above has those going away into a helper).

But then the actual data that expresses the content of the test case has "1" and "2", because it needs to be expressed as user IDs.

I think either way of naming will work; it just needs to be consistent, so that the reader doesn't have to translate back and forth in their head in order to understand the test case. If you want "A" and "B", the user IDs can be handled by having something like
```dart
  const idA = 1;
  const idB = 2;
```
up at the group level.